examples/01-filter/distance-between-surfaces.py

```diff
@@ -87,12 +87,45 @@ def hill(seed):
 from scipy.spatial import KDTree
 
 tree = KDTree(h1.points)
-d, idx = tree.query(h0.points )
-h0["distances"] = d
-np.mean(d)
+d_kdtree, idx = tree.query(h0.points )
+h0["distances"] = d_kdtree
+np.mean(d_kdtree)
 
 ###############################################################################
 p = pv.Plotter()
 p.add_mesh(h0, scalars="distances", smooth_shading=True)
 p.add_mesh(h1, color=True, opacity=0.75, smooth_shading=True)
 p.show()
+
+
+###############################################################################
+# Exact closest distance
+# ++++++++++++++++++++++
+#
+# The :func:`pyvista.DataSet.find_closest_cell` filter can be used to determine
+# the point on a surface that is closest to a set of points.  This method finds
+# points inside cells that are closest to the supplied points.  This provides
+# the exact closest distance from one mesh to the other. In this case, the
+# distances are calculated from points of ``h0`` to the entire mesh of ``h1``.
```

Comment:
I wonder how obvious to laypeople is the distinction between
1. closest mesh point,
2. closest point within closest cell,
3. theoretical distance of abstract mathematical objects being represented by the meshes

Case in point: when we have a `pv.Sphere()` it's a triangulated version of a sphere. There are 3 distances: distance to closest point, distance to closest point on closest triangle, and distance to exact sphere.

Obviously we can't know anything about the third case in pyvista, I just got unsure how obvious the "obviously" is, when we talk about "exact" distance.

Reply:
I added a significant discussion about points 1 and 2.  I think 3 is probably out of scope here.  Now that we no longer use a Sphere as a mesh anywhere, I think point 3 becomes less of a stumbling block [in these exact examples anyway].

Reply:
Yeah, looks good, thank you.